### PR TITLE
8196743: jstatd doesn't see new Java processes inside Docker container

### PR DESCRIPTION
**Pull request by kevinjwalls**
This is a merge request to fix the issue where a running jstatd does not see JVMs in docker containers started since the jstatd itself started.

jstatd was using the set of temp directories gathered on startup to scan for VMs, by reading their perfdata file.  That includes existing running VMs in docker instances using tmp dirs such as /proc/PID/root/tmp/.  But it needs to re-evaluate that set of temp dirs continually to see new VMs in docker containers.  Rescanning also means it "forgets" about tmp dirs which have disappeared, for containers that have shutdown, and it will now not be scanning those directories forever.

Have been testing this manually, watching jps connecting to jstatd and show newly created JVMs in docker containers come and go.  Running existing tests OK so far.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8196743](https://bugs.openjdk.java.net/browse/JDK-8196743): jstatd doesn't see new Java processes inside Docker container

### Reviewers
 * [Serguei Spitsyn](https://openjdk.java.net/census#sspitsyn) (@sspitsyn - **Reviewer**) ⚠️ Review applies to 658a9e27255c6a9e7cb414437a0669684d118472
 * [Yasumasa Suenaga](https://openjdk.java.net/census#ysuenaga) (@YaSuenag - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/3908/head:pull/3908` \
`$ git checkout pull/3908`

Update a local copy of the PR: \
`$ git checkout pull/3908` \
`$ git pull https://git.openjdk.java.net/jdk pull/3908/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 3908`

View PR using the GUI difftool: \
`$ git pr show -t 3908`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/3908.diff">https://git.openjdk.java.net/jdk/pull/3908.diff</a>

</details>

### CODE REVIEW

File: src/jdk.internal.jvmstat/share/classes/sun/jvmstat/perfdata/monitor/protocol/local/LocalVmManager.java [hunk 1]
```diff
@@ -76,10 +76,9 @@ public LocalVmManager() {
      */
     public LocalVmManager(String user) {
         this.userName = user;
-
+        setTempDirs();
```
Comment (YaSuenag) [inline src/jdk.internal.jvmstat/share/classes/sun/jvmstat/perfdata/monitor/protocol/local/LocalVmManager.java:79]: Is this call necessary? It looks like we don't need to hold `tmpdirs` anymore. Can we remove `tmpdirs` from class member and can call `PerfDataFile.getTempDirectories()` directly at L138?

**Review by sspitsyn — APPROVED** [review]
Hi Kevin,
This looks okay to me modulo the question from Yasumasa.
Thanks,
Serguei

File: src/jdk.internal.jvmstat/share/classes/sun/jvmstat/perfdata/monitor/protocol/local/LocalVmManager.java (hunk 1 above)
Comment (kevinjwalls) [reply src/jdk.internal.jvmstat/share/classes/sun/jvmstat/perfdata/monitor/protocol/local/LocalVmManager.java:79]: Hi - right, it's not necessary, I almost took it out, but had a minor concern about leaving tmpdirs unset.
But removing tmpdirs as a member is fine also.  That is testing well, will push an update shortly.

**Review by YaSuenag — APPROVED** [review]
Looks good! Thank you for fixing!

**Comment by kevinjwalls** [comment]
(For the record, the tier1_serviceability tests run OK for me.)

**Pull request closed without merging**